index.html

```diff
@@ -0,0 +1,203 @@
+        company: "Krellian",
+        mailto: "ben@krellian.com",
+        companyURL: "https://krellian.com",
+        w3cid: "51527"
+      }],
+      github: "w3c/web-thing-protocol",
+      shortName: "web-thing-protocol",
+      xref: true,
+      group: "web-thing-protocol",
+      wgPublicList: "public-web-thing-protocol",
+      edDraftURI: "https://w3c.github.io/web-thing-protocol/",
+      latestVersion: "https://w3c.github.io/web-thing-protocol/"
+    };
+  </script>
+</head>
+
+<body>
+  <section id="abstract">
+    <p>This document defines a WebSocket sub-protocol called the Web Thing
+      Protocol, for monitoring and controlling connected devices over the
+      World Wide Web. The Web Thing Protocol is intended as a common protocol
+      for the <a href="https://www.w3.org/WoT/">Web of Things</a>, to enable
+      ad-hoc interoperability between
+      <a href="https://www.w3.org/TR/wot-architecture/#dfn-web-thing">
+        Web Things</a> and their
+      <a href="https://www.w3.org/TR/wot-architecture/#dfn-consumer">Consumers</a>.
+    </p>
+  </section>
+
+  <section id="sotd">
+  </section>
+
+  <section id="introduction" class="informative">
+    <h2>Introduction</h2>
+    <p>This document defines a common WebSocket [[WEBSOCKETS]] sub-protocol for
+      monitoring and controlling connected devices over the Web, to enable
+      ad-hoc interoperability on the Web of Things.</p>
+    <p>The <a href="https://www.w3.org/WoT/">Web of Things</a> (WoT) is a
+      collection of standardised technology building blocks that help provide
+      interoperability on the
+      <a href="https://en.wikipedia.org/wiki/Internet_of_things">Internet of
+        Things</a> (IoT). The WoT Thing Description specification
+      [[wot-thing-description11]] defines a metadata format for
+      <em>describing</em> the capabilities of &quot;Things&quot; (connected
+      devices) on the Web. The WoT Discovery specification [[wot-discovery]]
+      defines mechanisms for <em>discovering</em> Things on the Web, e.g. using
+      an HTTP URL. This specification complements those building blocks by
+      defining a common protocol for <em>communicating</em> with Things over the
+      Web.
+    </p>
+    <p>WoT Thing Descriptions are designed to be protocol-agnostic, so that it
+      is theoretically possible to directly describe any brownfield IoT device
+      using any existing IoT protocol. However, this unconstrained complexity
+      makes it almost impossible to implement a single WoT Consumer which is
+      guaranteed to be able to communicate with any Web Thing. The Web Thing
+      Protocol is therefore designed specifically for developers of greenfield
+      implementations who want to benefit from ad-hoc interoperability. That
+      means that any Consumer and Web Thing which implement the protocol
+      should be able to communicate with each other out of the box, without
+      device-specific customisation.</p>
```

Comment:
I believe this is the most "controversial" point of the proposed structure. I would prefer to see it rephrased before merging it. It seems that the whole point is partially based on the discussion that is happening in the Profile taskforce which, sadly, I didn't have time to completely follow. Please if you have any pointers that invalidate what I'm saying here, share some pointers to the relevant issue or mail. 

First, I am not sure that the hypothesis that "impossible to implement a single WoT Consumer" is widely shared in the WoT WG group, but ofc, since this is another CG, we can explicitly say that. However, this is one of the core assumptions of the whole WoT idea, and I would say one of the reasons why we don't have a profile spec ready. Second, the rationale presented for building this new protocol seems to focus solely on addressing this specific gap in the WoT specification family. I believe it’s important to highlight that this protocol holds intrinsic value as a novel web-based interaction framework between devices, beyond merely filling a gap.

Reply:
Thank you very much for responding @relu91, I really appreciate your feedback.

Firstly, I don't understand why the statement that it is "almost impossible to implement a single WoT Consumer which is guaranteed to be able to communicate with any Web Thing" should be controversial. There are at least six extension points defined in the architecture of the Web of Things (protocol bindings, payload bindings, security mechanisms, discovery mechanisms, link relations and semantic contexts). It is objectively impossible to implement a Consumer which supports all current and future possible extensions. Even within a single protocol my experience has been that there is a lot of ambiguity when describing non-trivial interaction affordances using a Thing Description alone, which makes implementing a general purpose WoT Consumer very challenging.

Having said all of that, these are really more arguments for Profiles (which can constrain all of those extension points) than for the Web Thing Protocol (which is mainly focused on one of them). In my mind the Web Thing Protocol has always been a potential back up plan for if Profiles never make it through standardisation (which is why the charter covers an HTTP sub-protocol as well as WebSockets), but my hope has always been that that won't be necessary. That's why I've started with the WebSocket sub-protocol deliverable, which I think we agree is useful regardless of whether Profiles exist.

I think you make a good point that the Web Thing Protocol is valuable in its own right, separate from any questions about the level of interoperability which the core WoT specifications can provide. Whilst we are focused on that first deliverable, I think it's fair to suggest we untangle those issues and focus more on the value that a common WebSocket sub-protocol can provide in its own right. I also agree that it's important to maintain a good relationship with the WoT Working Group, and I genuinely hope for this specification to complement the work we're doing there.

I have therefore re-worked the Abstract and Introduction to remove the discussion of interoperability issues, and focus on the utility that a WebSocket sub-protocol can provide. I have also removed the empty Security, Discovery and Links sections for now, since they are areas which could arguably be better addressed through a WebSocket-based profile in the future.

Please let me know what you think.